src/explicit_euler_integrator.cc

```diff
@@ -25,9 +25,9 @@ void ExplicitEulerIntegrator::step(std::shared_ptr<Flux> flux,
         #ifdef FLYFT_OPENMP
         #pragma omp parallel for schedule(static) default(none) firstprivate(timestep,mesh) shared(rho,j)
         #endif
-        for (int idx=0; idx < mesh.shape(); ++idx)
+        for (int idx=0; idx < mesh->shape(); ++idx)
             {
-            const auto rate = (j(idx)-j(idx+1))/mesh.step();
+            const auto rate = mesh->integrateSurface(idx,j(idx),j(idx+1))/mesh->volume(idx);
```

Comment:
Since `j` is a DataView, you should be able to use that overload
```suggestion
            const auto rate = mesh->integrateSurface(idx, j))/mesh->volume(idx);
```